pkg/cluster/cluster_info_scraper.go

```diff
@@ -115,6 +128,65 @@ func (s *ClusterScraper) GetAllNodes() ([]*api.Node, error) {
 	return s.GetNodes(listOption)
 }
 
+func (s *ClusterScraper) GetMachineSetToNodeUIDsMap(nodes []*api.Node) map[string][]string {
+	machineSetToNodeUIDs := make(map[string][]string)
+	if s.caClient == nil {
+		return machineSetToNodeUIDs
+	}
+
+	machineSetList := s.getCApiMachineSets()
+	if machineSetList != nil {
+		for _, machineSet := range machineSetList.Items {
+			machines := s.getCApiMachinesFiltered(machineSet.Name, metav1.ListOptions{
+				LabelSelector: metav1.FormatLabelSelector(&machineSet.Spec.Selector),
+			})
+			nodeUIDs := []string{}
+			for _, machine := range machines {
+				if machine.Status.NodeRef.Name != "" {
+					if uid := findNodeUID(machine.Status.NodeRef.Name, nodes); uid != "" {
+						nodeUIDs = append(nodeUIDs, uid)
+					}
+				}
+			}
+			machineSetToNodeUIDs[fmt.Sprintf("%s/%s", machineSetNodePoolPrefix, machineSet.Name)] = nodeUIDs
```

Comment:
Extract to a separate function that creates name for the node pool

Reply:
Suggested name format -
Nodepool-machineset-<machineset-name>-<cluster-name>
in lieu of
Nodepool-kubeturbo.io/machineset/"<machineset-name>-<cluster-name>